presto-main/src/main/java/io/prestosql/sql/planner/optimizations/HashGenerationOptimizer.java

```diff
@@ -240,7 +240,7 @@ public PlanWithProperties visitMarkDistinct(MarkDistinctNode node, HashComputati
         {
```

Comment:
Can you please format (wrap lines) commit message like:

```
Remove redundant hash computation

Currently hash value is recomputed for aggregation node for queries 
with Union and also if a mark distinct operation is performed on 
a bigint column when optimize_hash_generation session is enabled.
```

Also I don't fully understand the commit message. How was before your commit and how it is going to be after your commit? Can you please describe what do you change?

Reply:
This commit fixes the hash generation which is computed once again for certain pattern of queries like 

- If we have a query like `SELECT count(*) from (SELECT colA from tableA union all SELECT colB from tableB)` we currently compute the same hash at two places one at tableScan level (for both the tables) and other at post `RemoteExchangeNode` 

-  For queries like in  prestodb/presto#8024 the hash value is computed during tableScan and also in `Fragment3`

Reply:
The plan changes can be seen here https://gist.github.com/Praveen2112/1c3a4d51c1c8b155921285dd8ee146ae

Reply:
Thanks!